### DISCUSSION TITLE
Using non-stdlib imports

### DISCUSSION THREAD

**Discussion by nwunderly** [traefik/yaegi#1270] [Q&A] [answered]
Hello, had a question about library usage. Not a feature request or a bug, but this repo doesn't have discussions enabled.

Was looking to import types from my own code into an interpreter. Based on the example in the README that uses `fmt`, I did some source diving. I figured out that `stdlib.Symbols` is a map of module paths to the module contents, also as a map.

So I tried
```go
	stdlib.Symbols["github.com/nwunderly/disgo"] = map[string]reflect.Value{
		"Context": reflect.ValueOf((*disgo.Context)(nil)),
	}
```
where `disgo` is just a module containing the `Context` type I was trying to use. It's worth noting that this is my attempt to develop an "eval" command for a Go-based Discord bot, which is why I was attempting to import the Context type. I just wanted to define a function that takes one of these structs as a parameter, then pull it out of the interpreter and call it. I managed to get this working without custom imports, defining a function, evaluating it then running it with params in the main process.

My issue was in the custom import. I got varied panics and unhelpful errors with small tweaks to this, but it seems that the above code alone isn't enough to allow me to import a custom module. I was just looking for advice, since I seem to be missing something. Is there something else necessary to be able to add modules to the imports for an interpreter? I'd appreciate any suggestions or guidance.

\(I doubt it's necessary (and frankly my code right now is a mess), but my attempts at this can be found [here](https://github.com/nwunderly/disgo/blob/master/test/eval/eval.go).\)

Cheers!

**Comment by mvertes** [comment, marked as answer]
You should look at the `yaegi extract` command to automatically generate wrappers around the packages that you want to expose to yaegi. The starting point is `yaegi help extract`.
